splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/CachedOperation.java

```diff
@@ -110,6 +110,11 @@ public String prettyPrint(int indentLevel) {
         if (!isOpen)
             throw new IllegalStateException("Operation is not open");
         populateCache();
+        try {
+            source.init(SpliceOperationContext.newContext(activation));
```

Comment:
@jyuanca shouldn't we have called source.init() during the initialization of the CachedOperation? Why do we still need to init the source again? Thanks!

Reply:
populateCache() closes source, so it needs to be re-initalized and open again.

Reply:
Thanks @jyuanca !